tests/linearGaussian_npse_test.py

```diff
@@ -1,300 +1,226 @@
-from typing import List
+from dataclasses import dataclass
+from typing import Optional
 
 import pytest
 from torch import eye, ones, zeros
 from torch.distributions import MultivariateNormal
 
-from sbi import analysis as analysis
 from sbi import utils as utils
 from sbi.inference import NPSE
 from sbi.simulators import linear_gaussian
 from sbi.simulators.linear_gaussian import (
-    samples_true_posterior_linear_gaussian_mvn_prior_different_dims,
     samples_true_posterior_linear_gaussian_uniform_prior,
     true_posterior_linear_gaussian_mvn_prior,
 )
-from sbi.utils import BoxUniform
 
 from .test_utils import check_c2st, get_dkl_gaussian_prior
 
 
-# We always test num_dim and sample_with with defaults and mark the rests as slow.
-@pytest.mark.parametrize(
-    "sde_type, num_dim, prior_str, sample_with",
-    [
-        ("vp", 1, "gaussian", ["sde", "ode"]),
-        ("vp", 3, "uniform", ["sde", "ode"]),
-        ("vp", 3, "gaussian", ["sde", "ode"]),
-        ("ve", 3, "uniform", ["sde", "ode"]),
-        ("subvp", 3, "uniform", ["sde", "ode"]),
-    ],
-)
-def test_c2st_npse_on_linearGaussian(
-    sde_type, num_dim: int, prior_str: str, sample_with: List[str]
+@dataclass
+class NpseTestCase:
+    num_dim: int
+    prior_type: Optional[str]
+    sde_type: str
+    sample_with: Optional[str] = None
+    iid_method: Optional[str] = None
+
+    @property
+    def likelihood_shift(self):
+        return -ones(self.num_dim)
+
+    @property
+    def likelihood_cov(self):
+        return 0.3 * eye(self.num_dim)
+
+    @property
+    def x_o(self):
+        return zeros(self.num_dim)
+
+    @property
+    def prior_mean(self):
+        if self.prior_type in {"gaussian", None}:
```

Comment:
so `None` just defaults to Gaussian?

Reply:
Yeah, I believe this is how it was before